include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -130,6 +130,43 @@ __find_start_point(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_
     return _split_point_t<_Index>{*__res, __index_sum - *__res + 1};
 }
 
+template <typename _Rng1, typename _Rng2, typename _value_t_rng1 = oneapi::dpl::__internal::__value_t<_Rng1>,
+          typename _value_t_rng2 = oneapi::dpl::__internal::__value_t<_Rng2>>
+constexpr auto
+__can_use_ternary_op(int)
+    -> decltype(true ? std::declval<_value_t_rng1>() : std::declval<_value_t_rng2>(), std::true_type{})
+{
+    return {};
+}
+
+template <typename _T1, typename _T2>
+constexpr auto
+__can_use_ternary_op(...) -> std::false_type
+{
+    return {};
+}
+
+// This implementation of __assing_impl is required for performance optimisation
+template <typename _Rng1, typename _Rng2, typename _Rng3, typename _Index>
+std::enable_if_t<__can_use_ternary_op<_Rng1, _Rng2>().value, void>
```

Comment:
I think to use this SFINAE trick, you need to add something which could be interpreted as an integer as the argument. 

```suggestion
std::enable_if_t<__can_use_ternary_op<_Rng1, _Rng2>(0).value, void>
```